starter-code/src/App.js

```diff
@@ -1,18 +1,135 @@
+  state = {
+    isOnClick: false,
+    foods: [...foods],
+    currentList: [],
+    todaysFood: [],
+    inputNumber: 0,
+    totalCalorie: 0
+  };
+
+  addFood = newFood => {
+    const copy = [...this.state.foods];
+    copy.push(newFood);
+    this.setState({ foods: copy });
+  };
+  toggleForm = () => {
+    this.setState(state => ({
+      isOnClick: !state.isOnClick
+    }));
+  };
+  handleSearch = evt => {
+    evt.preventDefault(evt);
+    let searchList = [...this.state.foods];
+    if (evt.target.value !== "") {
+      searchList = searchList.filter(food =>
+        food.name.toUpperCase().includes(evt.target.value.toUpperCase())
+      );
+    } else {
+      searchList = [...this.state.foods];
+    }
+    this.setState({
+      currentList: searchList
+    });
+  };
+
+  componentDidMount() {
+    this.setState({ currentList: this.state.foods });
+  }
+
+  getInputValue = (evt, f) => {
+    evt.preventDefault(evt);
+    let foods = [...this.state.foods];
+    let index = foods.findIndex(item => item.name === f.name);
+    let updatedFood = foods[index];
+    console.log(updatedFood);
+    updatedFood.quantity = Number(evt.target.value);
+
+    let updatedList = [...foods, ...(foods[index] = updatedFood)];
+
+    console.log(updatedList);
+
+    this.setState({
+      currentList: updatedList
+    });
+  };
+
+  addFood = f => {
+    let foods = [...this.state.foods];
+    let foodsIndex = foods.findIndex(item => item.name === f.name);
+    console.log(foodsIndex);
+    let updatedTodaysFood = foods[foodsIndex];
+    console.log(updatedTodaysFood);
+
+    let list = [...this.state.todaysFood];
+
+    if (list.findIndex(item => item.name === f.name) === -1) {
+      list.push(updatedTodaysFood);
+      this.state.totalCalorie +=
+        updatedTodaysFood.calories * updatedTodaysFood.quantity;
+    }
+
+    let calories = 0;
+    list.forEach(l => {
+      calories += l.quantity * l.calories;
+      console.log("DANS LA BOUCLE", calories);
+    });
+    this.setState({
+      todaysFood: list,
+      totalCalorie: calories
+    });
+  };
+
   render() {
+    console.log(this.state.todaysFood);
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      <div className="div-container">
+        <h1 className="title">IronNutrition</h1>
+        <AddFoodForm status={this.state.isOnClick} clbk={this.addFood} />
+        <button className="button" onClick={this.toggleForm}>
+          Add New Food
+        </button>
+        <hr />
+        <SearchBar clbk={evt => this.handleSearch(evt)} />
+        <hr />
+        <div className="columns">
+          <div className="column">
+            {this.state.currentList.map((f, i) => (
```

Comment:
because you are doing a map on this.state.currentList the new food created with the form is not display.